### ISSUE TITLE
add function to send message without using msg_send! ?

### ISSUE THREAD

**Issue by jmquigs** [SSheldon/rust-objc#43]
I'm writing a library where I would like to hide the use of objc in the implementation.  In the library, I define a macro that internally uses `msg_send!` and `sel!`  I've managed to hide most objc dependencies using `$crate` references, but due to the way macro expansion works, I can't hide `msg_send!` and `sel!`, which means that the call site needs to have an `extern crate objc` to use them.

Here is the call site for my macro (notice the extern crate objc earlier):
https://github.com/jmquigs/rs-barfly/blob/master/examples/simple/src/main.rs#l24

And here is the macro definition:
https://github.com/jmquigs/rs-barfly/blob/master/src/lib.rs#l170
(notice how it uses `msg_send!`)

Is it feasible to export a `send_msg()` function from objc so that `msg_send!` can be bypassed?  I looked at using "expanded" code directly, but quickly ran into dependencies on `MessageArguments` and (worse) the associated platform modules.  So this may be non-trivial to implement.  

Thanks in advance

**Comment by SSheldon** [comment]
Hey @jmquigs, to hide the usage of the msg_send! macro you could put it inside of a function in your crate that your macro calls. For example, you could make a function like:
``` rust
fn set_item_target(item: id, target: id) {
    msg_send![item, setTarget:target]
}
```

And now in your macro, just call `$crate::set_item_target(item, &mut *cb_obj)` instead of using `msg_send!`.

For some more general feedback, it looks like you don't actually need a macro here! I don't see a reason why you need a different class for each fly item; instead you could just have one class, say "FlyCallback", and its selector always has the same name. Those are the only idents passed, so it looks like the macro could then become a function! (I just gave a quick look over the code, I might be mistaken...)

Hope this helps!

**Comment by SSheldon** [comment]
I realize that suggestion may have been hard to follow, I opened a pull request to describe what I meant: jmquigs/rs-barfly#1.
